src/components/pagination/pagination.tsx

```diff
@@ -59,10 +64,48 @@ export const EuiPagination: FunctionComponent<Props> = ({
   activePage = 1,
   onPageClick = () => {},
   compressed,
+  'aria-controls': ariaControls,
   ...rest
 }) => {
-  const classes = classNames('euiPagination', className);
+  const safeClick = (e: MouseEvent, pageIndex: number) => {
+    e.preventDefault();
+
+    if (ariaControls) {
+      const controlledElement = document.getElementById(ariaControls);
+
+      if (controlledElement) {
+        controlledElement.focus();
+      }
+    }
+
+    onPageClick(pageIndex);
+  };
+  const PaginationButton = ({
+    pageIndex,
+    inList = true,
+  }: {
+    pageIndex: number;
+    inList?: boolean;
+  }) => {
+    const button = (
+      <EuiPaginationButton
+        isActive={pageIndex === activePage}
+        totalPages={pageCount}
+        onClick={(e: MouseEvent) => safeClick(e, pageIndex)}
+        pageIndex={pageIndex}
+        {...hasControl && { 'aria-controls': ariaControls }}
+        hideOnMobile
+      />
+    );
+
+    if (inList) {
+      return <li className="euiPagination__item">{button}</li>;
+    }
 
+    return button;
+  };
+  const classes = classNames('euiPagination', className);
+  const hasControl = typeof ariaControls === 'string';
```

Comment:
EUI code pattern prefers eliminating the `undefined` option instead of `typeof`ing

```suggestion
  const hasControl = ariaControls !== undefined;
```